### PR TITLE
fix(streams): saturate GauntletStream step_count at int32 max

### LINKED ISSUE #1843: streams: GauntletStream.step step_count wraps negative and resets the segment schedule at int32 max
## Summary

`GauntletStream.step` advances its lifetime `step_count` with a bare
`state.step_count + 1` (`alberta_framework/streams/gauntlet.py:422`). The
counter is declared `dtype=jnp.int32` (`gauntlet.py:369`) and there is no
saturating increment on this stream. At `step_count == 2**31 - 1` the next
step wraps the clock to `INT32_MIN`.

`step_count` drives the segment schedule via
`segment_of(state.step_count)` (`gauntlet.py:377`, `401-403`):
`seg = step // segment_length` clamped to the last segment. A wrapped
negative step makes `step // segment_length` negative, and the clip resets
the course to segment 0 — the gauntlet silently restarts its segment
schedule after lifetime exhaustion instead of staying on the last segment.
Because `seg` also selects the drift/scale behavior (`gauntlet.py:406-432`)
and the linear readout weights, the produced features stop matching the
course's post-exhaustion behavior.

This is the same counter-wrap class already fixed elsewhere in the tree
(`learners.py` #1372, `synthetic.py` #1366, `feature_discovery.py` #1368,
`alberta_plan_step1.py` #1370, `horde.py` #1787, `reward_model.py` #1017,
`upgd.py` #1841); `GauntletStream` was not covered. Note the sibling
`LifetimeGauntletStream` already uses a checked saturating words increment
and is not affected.

## Reproduction

```python
import jax.numpy as jnp
import jax.random as jr
from alberta_framework.streams.gauntlet import GauntletStream, NUM_SEGMENTS

stream = GauntletStream()
state = stream.init(jr.key(0)).replace(
    step_count=jnp.asarray(2**31 - 1, dtype=jnp.int32)
)
seg_at_max = stream.segment_of(state.step_count)          # 8 (last)
wrap = jnp.asarray(2**31 - 1, dtype=jnp.int32) + 1
seg_at_wrap = stream.segment_of(wrap)                     # 0 (restart)
print(int(seg_at_max), int(seg_at_wrap))  # 8 0
```

`seg at INT32_MAX: 8`, `seg at wrap INT32_MIN: 0` — the schedule resets to
segment 0 instead of saturating on the last segment.

## Fix

Replace the bare increment at `gauntlet.py:422` with a saturating int32
increment (floor at 0, cap at `INT32_MAX - 1`, then `+ 1`), consistent with
`normalizers._saturating_int32_counter_increment` and the other
lifetime-clock fixes. Add a regression test that plants the exhausted clock,
runs one `step`, and asserts the clock saturates and the segment stays on
the last segment.

Fix incoming in a linked PR with the reproduction test.

### PR DESCRIPTION
**Pull request by deepanshu-yd**
Closes #1843

## Problem

`GauntletStream.step` advanced the lifetime `step_count` with a bare `state.step_count + 1` (`alberta_framework/streams/gauntlet.py:422`). The counter is declared `dtype=jnp.int32` and has no saturating increment. At `step_count == 2**31 - 1` the next step wraps the clock to `INT32_MIN`.

`step_count` drives the segment schedule via `segment_of` (`step // segment_length` clamped to the last segment). A wrapped negative step makes the segment reset to 0 — the gauntlet silently restarts its course instead of staying on the last segment — which also flips the drift/scale/readout behavior selected per segment.

## Fix

Replace the bare increment with the tree-wide saturating int32 counter increment (`normalizers._saturating_int32_counter_increment`), matching the existing lifetime-clock fixes in learners, synthetic streams, feature discovery, Step 1, horde, reward_model, and upgd.

## Tests

Adds `tests/test_gauntlet_lifetime_clock.py`: plants the exhausted clock and asserts the counter saturates at `INT32_MAX` and the segment stays on the last segment.

- `pytest tests/test_gauntlet_*.py`: 18 passed
- `ruff check`: clean
- `mypy`: clean

### CODE REVIEW

**Comment by lalalune** [comment]
Incorporated and validated in #1874, which preserved this contribution in the consolidated history and has now merged to main. Closing this source PR as resolved.
